src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/IndexQueueService.kt

```diff
@@ -14,11 +17,26 @@ import org.springframework.stereotype.Service
 class IndexQueueService(
   @Autowired @Qualifier("awsSqsIndexASyncClient") private val awsSqsIndexASyncClient: AmazonSQSAsync,
   @Autowired @Qualifier("indexQueueUrl") private val indexQueueUrl: String,
-  private val gson: Gson
-) {
+  private val gson: Gson,
+  private val telemetryClient: TelemetryClient)
+{
 
   fun sendIndexRequestMessage(payload: PrisonerIndexRequest) {
-    awsSqsIndexASyncClient.sendMessageAsync(SendMessageRequest(indexQueueUrl, gson.toJson(payload)))
+    awsSqsIndexASyncClient.sendMessageAsync(SendMessageRequest(indexQueueUrl, gson.toJson(payload)),
+      object : AsyncHandler<SendMessageRequest?, SendMessageResult?> {
```

Comment:
This is only raising an event when message failed to go on the queue - I really want to if the actual message is not processed successfully.